packages-exp/auth-exp/src/core/auth/auth_impl.ts

```diff
@@ -107,17 +108,21 @@ export class AuthImpl implements Auth, _FirebaseService {
         return;
       }
 
-      await this.initializeCurrentUser();
+      await this.initializeCurrentUser(popupRedirectResolver);
 
       if (this._deleted) {
         return;
       }
 
       this._isInitialized = true;
-      this.notifyAuthListeners();
     });
 
-    return this._initializationPromise;
+    // After initialization completes, throw any error caused by redirect flow
+    return this._initializationPromise.then(() => {
```

Comment:
why then instead of await?

Reply:
If we await it, the error becomes part of the control flow of initialization. If there's an error with redirect during initialization, we still want auth to finish initializing

Reply:
shouldn't we just wrap in a try catch then?